third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -450,28 +462,50 @@ class BlockedToMFMA : public OpRewritePattern<tt::DotOp> {
     auto oldBType = cast<RankedTensorType>(b.getType());
     auto ctx = oldAType.getContext();
 
-    ttg::AMDMfmaEncodingAttr mfmaEnc;
+    Type aElemType = oldAType.getElementType();
+    Type bElemType = oldBType.getElementType();
+    bool withScale =
+        mfmaVersion == 4 && isF8F6F4(aElemType) && isF8F6F4(bElemType);
+
+    // If mfmaVersion == 4 and both inputs are of F8F6F4 types, we will try to
+    // use the V_MFMA_*_F8F6F4 instructions since it has higher bandwidth. If we
+    // can't find a proper instruction, we will fall back to select from normal
+    // mfma instructions.
+    MfmaIntrinsic mfmaInstr;
```

Comment:
Nit: no need to create a local `mfmaInstr` here which would require copy to it later. You can just name `maybeMfmaInstr` as `mfmaInstr` and use `->` to access its content.

Reply:
Emmmm I got compilation error saying `FailureOr<MfmaIntrinsic>` can't be reassigned thus I changed to `MfmaIntrinsic`. But it worked now. Maybe it's influenced by something else.